### PR TITLE
Fix cursor placement in stdin history search and navigation

### PR DESCRIPTION
**Pull request by krassowski**
## References

Fixes https://github.com/jupyterlab/jupyterlab/issues/14222

## Code changes

Ensure the cursor is placed at the end by:
- preventing default action of the keydown event
- setting the selection at the end

## User-facing changes

Cursor stays at the end when using <kbd>arrow up</kbd> key

## Backwards-incompatible changes

None, but will require a manual backport

### CODE REVIEW

File: packages/outputarea/src/widget.ts [hunk 1]
```diff
@@ -1151,6 +1157,13 @@ export class Stdin extends Widget implements IStdin {
     this._input.removeEventListener('keydown', this);
   }
 
+  private _setInputValue(value: string) {
+    this._input.value = value;
+    // Set cursor at the end; this is usyally not necessary when input is
```
Comment (mlucool) [inline packages/outputarea/src/widget.ts:1162]: Typo: usyally

**Review by fcollonval — APPROVED** [review]
Thanks @krassowski

**Comment by fcollonval** [comment]
@meeseeksdev please backport to 3.6.x

**Comment by fcollonval** [comment]
@krassowski I tried backporting it manually. But it seems not so straightforward because https://github.com/jupyterlab/jupyterlab/pull/13937 was not backported. Should #13937 and this one be backported or only this one?

**Comment by krassowski** [comment]
https://github.com/jupyterlab/jupyterlab/pull/13937 is a new feature and I think it could only be backported if we decide to release 3.7. I will open a backport PR for this one.

**Pull request merged**